runtime/testy/runtime.py

```diff
@@ -47,11 +48,14 @@ def runtime():
       print("Starting studentCode attempt: %s" % (restartCount,))
       spawnProcess(PROCESS_NAMES.STUDENT_CODE, runStudentCode)
       while True:
-        newBadThing = badThingsQueue.get(block=True)
```

Comment:
This code is duplicated with `runtimeTest`... and this PR only fixes the bugs in the `runtimeTest` version. You should at least keep the two in sync, and preferably de-duplicate the code asap.